### ISSUE TITLE
getLoggedValues fails when logging wrapper does not log x

### ISSUE THREAD

**Issue by mb706** [jakobbossek/smoof#143]
```r
library("smoof")
fn = makeSphereFunction(dimensions = 2L)
fn = addLoggingWrapper(fn, logg.x = FALSE) # (the default)
res = optim(fn, par = c(1, 1), method = "Nelder-Mead")
log.res = getLoggedValues(fn)       
#> Error in `colnames<-`(`*tmp*`, value = getParamIds(ParamHelpers::getParamSet(fn),  : 
#>   attempt to set 'colnames' on an object with less than two dimensions
```
The problem seems to be this line:
https://github.com/jakobbossek/smoof/blob/d65247258fab57d08a5a76df858329a25c0bb1b8/R/getLoggedValues.R#L28

which accesses a variable that is not created when `logg.x` is `FALSE`.

I also have the impression that this line could cause trouble at some point:
https://github.com/jakobbossek/smoof/blob/d65247258fab57d08a5a76df858329a25c0bb1b8/R/addLoggingWrapper.R#L91

**Comment by mb706** [comment]
Another problem occurs when `fn` is then called with a list of values, which I assume should be allowed? The logging wrapper gives a warning in this case and logs the wrong values.

```r
library("smoof")
fn = makeSphereFunction(dimensions = 2L)
fn(list(x = c(1, 0)))  # works okay
#> [1] 2
fn = addLoggingWrapper(fn, logg.x = TRUE)
fn(list(x = c(1, 0)))  # gives a warning
#> [1] 2
#> Warning message:
#> n `[<-.data.frame`(`*tmp*`, curr.idx, , value = list(x = c(1, 1 :
#>   replacement element 1 has 2 rows to replace 1 rows
getLoggedValues(fn)  # wrong! $pars should be 1, 0! Instead, x1 is repeated twice
#> $pars
#>   x1 x2
#> 1  1  1
#>
#> $obj.vals
#> [1] 1
```

**Comment by jakobbossek** [comment]
Thanks for the bug report! I will take care of this ASAP.

**Comment by jakobbossek** [comment]
Fixed via [3e2efee](https://github.com/jakobbossek/smoof/commit/3e2efeee204462e9b6bf46d957b52d7beee213f2). I added two new test cases (`logg.x = TRUE` and `logg.y = FALSE` and the vice versa) and modified `addLoggingWrapper` and `getLoggedValues`. GitHub workflows currently fail. I guess this is due to deprecated workflow version, but I do not the time now to handle this.

Thanks again for reporting! 👏🏼 

Regarding your other questions: passing a list works well if the function signature is not *simple*, i.e., the Boolean *has.simple.signature*  is set to *FALSE* in the smoof function constructor. However, this is usually the case for mixed parameter functions.
